x-pack/plugins/fleet/server/saved_objects/model_versions/outputs.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectModelDataBackfillFn } from '@kbn/core-saved-objects-server';
+
+import type { Output } from '../../../common';
+
+export const backfillOutputPolicyToV7: SavedObjectModelDataBackfillFn<
+  Output & {
+    topics?: Array<{
+      topic: string;
+      when?: {
+        type?: string;
+        condition?: string;
+      };
+    }>;
+  },
+  Output
+> = (outputDoc) => {
+  if (outputDoc.attributes.type === 'kafka' && outputDoc.attributes.topics) {
+    if (!outputDoc.attributes.topic) {
+      outputDoc.attributes.topic = outputDoc.attributes.topics?.filter((t) => !t.when)?.[0]?.topic;
+    }
+
+    delete outputDoc.attributes.topics;
```

Comment:
The fact that we're removing this field makes it non-BWC. I wonder if we can perform the migration in 2 steps:
1. Maintains both fields
2. (In a couple of weeks) Deletes the deprecated `topics` array

Reply:
@afharo since kibana 8.13 we already handle the fact that `topics` could not be here and the `topic` field is present, so I think we should not have any BWC issues

Reply:
let me know if you think this could still be an issue

Reply:
oh! if 8.13+ already uses the single `topic` field, then all good on my end. Approving